### PR TITLE
feat: Deploy getters and tests

### PR DESCRIPTION
**Pull request by deluca-mike**

### CODE REVIEW

File: test/Deploy.t.sol [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    function test_deploy() external {
+        (address minterGateway_, address minterRateModel_, address earnerRateModel_) = deploy(
+            address(this),
+            2,
+            address(_ttgRegistrar)
+        );
+
+        address mToken_ = getExpectedMToken(address(this), 2);
+
+        // Minter Gateway assertions
```
Comment (PierrickGT) [inline test/Deploy.t.sol:35]: We can also assert the `rateModel()`.

File: test/Deploy.t.sol [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        assertEq(minterGateway_, getExpectedMinterGateway(address(this), 2));
+        assertEq(IMinterGateway(minterGateway_).ttgRegistrar(), address(_ttgRegistrar));
+        assertEq(IMinterGateway(minterGateway_).ttgVault(), _TTG_VAULT);
+        assertEq(IMinterGateway(minterGateway_).mToken(), mToken_);
+
+        // MToken assertions
```
Comment (PierrickGT) [inline test/Deploy.t.sol:41]: We can also assert the `rateModel()`.

**Review by PierrickGT — CHANGES_REQUESTED** [review]

File: test/Deploy.t.sol (hunk 1 above)
Comment (deluca-mike) [reply test/Deploy.t.sol:35]: No, becauss those only take effect and TTG proposals are executed.

File: test/Deploy.t.sol (hunk 2 above)
Comment (deluca-mike) [reply test/Deploy.t.sol:41]: No, because those only take effect and TTG proposals are executed.

**Review by PierrickGT — APPROVED** [review]

**Review by toninorair — APPROVED** [review]

**Pull request merged**